c7n/resources/ebs.py

```diff
@@ -63,13 +104,18 @@ def process(self, snapshots):
          # Be careful re image snapshots, we do this by default
         # to keep things safe by default, albeit we'd get an error
         # if we did try to delete something associated to an image.
-        if self.data.get('skip-ami-snapshots', True):
-            # Auto filter ami referenced snapshots, build map
-            c = local_session(self.manager.session_factory).client('ec2')
-            for i in c.describe_images(Owners=['self'])['Images']:
-                for dev in i.get('BlockDeviceMappings'):
-                    if 'Ebs' in dev and 'SnapshotId' in dev['Ebs']:
-                        snaps.add(dev['Ebs']['SnapshotId'])
+        
+        #=======================================================================
```

Comment:
we'll still need this filtering in the action, to prevent accidental usage sans filter that would cause exceptions, i'd like to encapsulate the retrieval here into a private utility function that both the filter and action use.